src/Core/Types/TypeBase.cs

```diff
@@ -12,34 +15,45 @@ namespace Axe.Windows.Core.Types
     /// </summary>
     public abstract class TypeBase
     {
-        const string NamePattern = "UIA_";
+        const string DefaultNamePattern = "UIA_";
         protected Dictionary<int, string> Dic { get; private set; }
+        private readonly string _namePattern = null;
 
-        protected TypeBase() : this(NamePattern) { }
+        protected TypeBase() : this(DefaultNamePattern) { }
 
         protected TypeBase(string np)
         {
+            _namePattern = np;
```

Comment:
I have mixed feelings about storing _namePattern as a class member. It's only needed during the constructor, so this could just as easily be handled as a parameter.

Reply:
I had those same mixed feelings. But I chose the member variable so that I didn't have to keep passing the value around as a parameter.